### ISSUE TITLE
Use of `and` with configuration `scope=given`

### ISSUE THREAD

**Issue by matteo-cristino** [dyne/Zenroom#764]
When using the configuation
```
scope=given
```
With a contract in which a `when` statement start with a `and`, e.g.:
```gherkin
Given I have a 'integer' named 'percentage'

When I set 'modulo' to '100' as 'integer'
and I create the array of '1' random numbers modulo 'modulo'

Then print the 'array'
```
Then the `when` statement is ignored while the `and` statement is taken as a `given` one, thus the following error is returned
```
[!]  Zencode parser error
[!]  [source 'zencode']:213: Zencode line 5 found invalid statement after a valid one in the given phase: and I create the array of '1' random numbers modulo 'modulo'
```

The same thing happens with an `and` in the `then` phase

**Comment by jaromil** [comment]
This is tricky to solve: I think we need to create a special When function callback which records the statements also when `scope=given` rather than ignoring them all.

**Comment by jaromil** [comment]
solved in #765 ! thanks @matteo-cristino for the simple solution
